### PR TITLE
enable taskset core pinning in addition to numactl

### PR DESCRIPTION
**Pull request by min-jean-cho**
- port https://github.com/intel-innersource/frameworks.ai.pytorch.ipex-cpu/pull/740 to `run_cpu`
- use-case by https://github.com/pytorch/serve/pull/2166 where `numactl` is unavailable (e.g., requires `privileged` mode) 

This PR automatically tries taskset if numactl core binding doesn't work.

Reference: 
`taskset` is added to adapt to launcher use-cases such as in docker where `numactl` requires to be ran in  `privileged` mode, where the  `privileged` mode "wont work for deployments like sagemaker for example" as raised by TorchServe. Please see [torchserve ipex docker discussion](https://github.com/pytorch/serve/pull/1401#issuecomment-1090817704) for reference. To address such use-cases, `taskset` can be used in place of `numactl` to set core affinity. Note that, unlike `numactl`, `taskset` does not provide memory binding to local memories; however, memory binding may not be needed in these use-cases  that typically do not span multi sockets. Hence we can automatically try taskset if numactl doesn't work.

### CODE REVIEW

File: torch/backends/xeon/run_cpu.py
Comment (jingxu10) [inline torch/backends/xeon/run_cpu.py:289]: please redirect the output to /dev/null. We don't want to print result of this test command out.

Comment (min-jean-cho) [reply torch/backends/xeon/run_cpu.py]: Thanks @jingxu10, done.

**Comment by jingxu10** [comment]
@malfet @jgong5 would you take a review?
Thank you.

**Review by jgong5 — APPROVED** [review]
Consider to copy the description from https://github.com/intel-innersource/frameworks.ai.pytorch.ipex-cpu/pull/740 to the description here? It is a private link. Not everyone can access.

**Comment by min-jean-cho** [comment]
Hi @malfet, could you help review this PR? Looks like it needs at least one approval from the Core Maintainers. Thanks!

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**
